tap_pendo/streams.py

```diff
@@ -446,7 +452,10 @@ def sync(self, state, start_date=None, key_id=None):
         if stream_response and sub_stream and sub_stream.is_selected():
             self.sync_substream(state, self, sub_stream, stream_response)
 
-        update_currently_syncing(state, None)
+            # Get parent data again as stream_response returned from request() is a generator
+            # which flush out during sync_substream call above
+            stream_response = self.request(self.name, json=self.get_body()) or []
```

Comment:
In python, there is no way to clone the generator so we have to make a request again for parent data.  
Reference: https://stackoverflow.com/questions/29835784/copy-a-generator